### PR TITLE
Integrity check for DIFF mode

### LINKED ISSUE #1226: Integrity tests to check the import-diff flow
While the Integrity tests check the `import-osm` flow, they do not test the `import-diff` flow. As a result, the SQL code that performs table updates is not tested.

[Replace materialized view of water by tables with diff update #853](https://github.com/openmaptiles/openmaptiles/pull/853) and similar PRs have continuously increased the complexity of the code that is used only in flows based on `import-diff`. Currently, this code is not tested by the integrity tests, including for typing errors.

Such a test requires finding a source changefile that performs modifications in all OMT tables for a relatively small area.

It would be nice to also add performance measurements for the flow.

Related to comments for b011b27e52726361d352932ad923de7e05893cc4

### PR DESCRIPTION
**Pull request by zstadler**
This test applies 2-3 months of weekly updates on the `europe/monaco` extract from [Geofabrik](http://download.geofabrik.de/europe/).

It is worth noting that the contents of the updates may vary, and some SQL update flows may not be tested if the relevant changes did not occur during that period.

Resolves #1226

### CODE REVIEW

**Comment by ZeLonewolf** [comment]
I was thinking about your note about update variability, and I'm wondering if it makes sense to start creating a library of fake sample data for the purpose of creating static unit tests.

**Comment by zstadler** [comment]
That's not an easy task...

**Comment by ZeLonewolf** [comment]
Rome wasn't built in a day :)

What I'm protesting wouldn't be a replacement for this PR, it would be in addition to it.

File: .github/workflows/integrity.yml [hunk 1]
```diff
@@ -36,3 +36,36 @@ jobs:
...
+          docker-compose run --rm --user=$(id -u):$(id -g) openmaptiles-tools sh -c "
+            while ! osmupdate --keep-tempfiles --base-url=$(sed -n 's/ *\"replication_url\": //p' data/$area.repl.json) data/$area.osm.pbf data/changes.osc.gz ; do 
+              sleep 2;
+              echo Sleeping...;
+              sleep 630;
+            done"
+          echo Downloading updates completed
+          echo Importing updates
+          make import-diff
+          echo Generaing new tiles
```
Comment (ZeLonewolf) [inline .github/workflows/integrity.yml:70]: Typo, should be "Generating"

Comment (zstadler) [reply .github/workflows/integrity.yml]: Thanks @ZeLonewolf.
 Corrected.

**Comment by zstadler** [comment]
Yes. When someone creates the first static unit test, it would be a separate PR.

**Comment by TomPohys** [comment]
This is a great improvement in CI testing. Thank you very much!

**Review by TomPohys — APPROVED** [review]

**Pull request merged**